executor/metric_reader.go

```diff
@@ -36,14 +38,14 @@ const promReadTimeout = time.Second * 10
 
 // MetricRetriever uses to read metric data.
 type MetricRetriever struct {
-	table      *model.TableInfo
-	tblDef     *metricschema.MetricTableDef
-	outputCols []*model.ColumnInfo
-	retrieved  bool
+	table     *model.TableInfo
+	tblDef    *metricschema.MetricTableDef
+	extractor *plannercore.MetricTableExtractor
+	retrieved bool
 }
 
-func (e *MetricRetriever) retrieve(ctx context.Context, sctx sessionctx.Context) (fullRows [][]types.Datum, err error) {
-	if e.retrieved {
+func (e *MetricRetriever) retrieve(ctx context.Context, sctx sessionctx.Context) (partRows [][]types.Datum, err error) {
```

Comment:
The named return parameters make confused. The `partRows` is a named return parameter and it should be used to indicate the final rows, but it is used as a local variable.

Reply:
Oh, sorry, it's a mistake.